indigo_api/serializers.py

```diff
@@ -389,8 +394,11 @@ def update_document(self, document, validated_data=None):
         # Document content must always come first so it can be overridden
         # by the other properties.
         content = validated_data.pop('content', None)
+        # TODO: include provision_eid
+        provision_eid = validated_data.pop('provision_eid', None)
         if content is not None:
-            # TODO: this is where we want to intercept the XML update (as well?)
+            if provision_eid:
+                content = document.update_provision_xml(provision_eid=provision_eid, provision_xml=content)
```

Comment:
I think `update_provision_xml` should call `reset_xml` itself -- so it updates the full internal representation of the document.